### PR TITLE
Do not allow dot in namespace and hostname

### PR DESCRIPTION
**Pull request by eduardoveiga**
- [x] ensure API do not allow wrong namespace and hostname - @eduardoveiga 
- [x] migrate old namespace and hostname replacing `.` to `-` - @eduardoveiga 
- [x] validate the hostname and namespace on UI to avoid sending wrong data - @noreng-jg

### CODE REVIEW

File: api/store/mongo/migrations/migration_23.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+var migration_23 = migrate.Migration{
+	Version:     23,
+	Description: "change dot in namespace name and hostname to -",
+	Up: func(db *mongo.Database) error {
+		logrus.Info("Applying migration 23 - Up")
+		result, err := db.Collection("namespaces").UpdateMany(context.TODO(), bson.D{}, []bson.M{
+			{
+
+				"$set": bson.M{"name": bson.M{
+					"$replaceOne": bson.M{"input": "$name", "find": ".", "replacement": "-"},
```
Comment (otavio) [inline api/store/mongo/migrations/migration_23.go:24]: What happens if there are multiple `.` on the field?

Comment (eduardoveiga) [reply api/store/mongo/migrations/migration_23.go]: I changed to "replaceAll" and now works with multiple dots

File: api/store/mongo/migrations/migration_23.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
+package migrations
+
+import (
+	"context"
+
+	//"github.com/shellhub-io/shellhub/pkg/models"
```
Comment (otavio) [inline api/store/mongo/migrations/migration_23.go:6]: Please remove the dead code from here.

**Review by otavio — CHANGES_REQUESTED** [review]

**Review by otavio — APPROVED** [review]

**Review by otavio — CHANGES_REQUESTED** [review]
@eduardoveiga please fix the comments on the migration so we can consider this for merging.

File: api/store/mongo/migrations/migration_23.go (hunk 2 above)
Comment (eduardoveiga) [reply api/store/mongo/migrations/migration_23.go]: done

**Review by otavio — APPROVED** [review]

**Pull request merged**